commit c11a82b62aefc21e070c527f59a1f9c57a7b4f36
Author: Richard Jones <rjones@redhat.com>
Date:   Thu Dec 10 16:39:07 2009 +0000

    qemu driver: Fix segfault in libvirt/libvirtd when uri->path is NULL.
    
    See also:
    https://bugzilla.redhat.com/show_bug.cgi?id=545400#c1

diff --git a/src/qemu/qemu_driver.c b/src/qemu/qemu_driver.c
index 2fb059d..e9cc8c3 100644
--- a/src/qemu_driver.c
+++ b/src/qemu_driver.c
@@ -2651,6 +2651,15 @@ static virDrvOpenStatus qemudOpen(virConnectPtr conn,
             return VIR_DRV_OPEN_ERROR;
         }
 
+        if (conn->uri->path == NULL) {
+            qemudReportError(conn, NULL, NULL, VIR_ERR_INTERNAL_ERROR,
+                             _("no QEMU URI path given, try %s"),
+                             qemu_driver->privileged
+                               ? "qemu:///system"
+                               : "qemu:///session");
+                return VIR_DRV_OPEN_ERROR;
+        }
+
         if (qemu_driver->privileged) {
             if (STRNEQ (conn->uri->path, "/system") &&
                 STRNEQ (conn->uri->path, "/session")) {
